fabtests/prov/efa/src/efa_exhaust_mr_reg_common.c

```diff
@@ -5,19 +5,39 @@
 int ft_efa_open_ibv_device(struct ibv_context **ctx) {
 	int num_dev = 0;
 	struct ibv_device **dev_list;
+	struct efadv_device_attr efadv_attr = {0};
+	int ret;
 
 	dev_list = ibv_get_device_list(&num_dev);
 	if (num_dev < 1) {
 		FT_ERR("No ibv devices found");
 		ibv_free_device_list(dev_list);
 		return -1;
-	} else if (num_dev > 1) {
-		FT_WARN("More than 1 ibv devices found! This test will only "
-			"exhaust MRs on the first device");
 	}
-	*ctx = ibv_open_device(dev_list[0]);
+
+	/* Loop through devices and find the first EFA device */
+	for (int i = 0; i < num_dev; i++) {
+		*ctx = ibv_open_device(dev_list[i]);
+		if (!*ctx) {
+			FT_ERR("cannot open device %d", i);
```

Comment:
Should it be `FT_ERR` if the failure to open device is expected? I would change to `FT_WARN`

Reply:
I added my comment in the wrong place